src/libraries/System.IO.FileSystem/tests/File/OpenHandle.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    // to avoid a lot of code duplication, we reuse FileStream tests
+    public class File_OpenHandle : FileStream_ctor_options_as
+    {
+        protected override string GetExpectedParamName(string paramName) => paramName;
+
+        protected override FileStream CreateFileStream(string path, FileMode mode)
+        {
+            FileAccess access = mode == FileMode.Append ? FileAccess.Write : FileAccess.ReadWrite;
+            return new FileStream(File.OpenHandle(path, mode, access, preallocationSize: PreallocationSize), access);
+        }
+
+        protected override FileStream CreateFileStream(string path, FileMode mode, FileAccess access)
+            => new FileStream(File.OpenHandle(path, mode, access, preallocationSize: PreallocationSize), access);
+
+        protected override FileStream CreateFileStream(string path, FileMode mode, FileAccess access, FileShare share, int bufferSize, FileOptions options)
+            => new FileStream(File.OpenHandle(path, mode, access, share, options, PreallocationSize), access, bufferSize, (options & FileOptions.Asynchronous) != 0);
+
+        [Fact]
+        public override void NegativePreallocationSizeThrows()
+        {
+            ArgumentOutOfRangeException ex = Assert.Throws<ArgumentOutOfRangeException>(
```

Comment:
In this case it would be "value", as it's thrown by the setter here:

https://github.com/dotnet/runtime/blob/b90631f64809a77beb7a6aef787f0254956d522b/src/libraries/System.Private.CoreLib/src/System/IO/FileStreamOptions.cs#L99

So I would prefer not to validate it

Reply:
> So I would prefer not to validate it

Because we don't like the name that emerges in the exception?  Does that instead suggest we're throwing the wrong thing from this API?